src/js/components/generateFiles/GenerateFilesOverlay.jsx

```diff
@@ -7,13 +7,48 @@ import React from 'react';
 import CommonOverlay from '../SharedComponents/overlays/CommonOverlay.jsx';
 import * as Icons from '../SharedComponents/icons/Icons.jsx';
 import LoadingBauble from '../SharedComponents/overlays/LoadingBauble.jsx';
+import * as ReviewHelper from '../../helpers/reviewHelper.js';
 
 const defaultProps = {
 	state: 'incomplete'
 };
 
 export default class GenerateFilesOverlay extends React.Component {
 
+	constructor(props){
+		super(props);
+	}
+
+	componentDidMount(){
+		if (this.props.submissionID !== null) {
+            ReviewHelper.fetchStatus(this.props.submissionID)
+                .then((data) => {
+                    data.ready = true;
+                    this.setState({'agency_name':data.agency_name})
+                })
+                .catch((error) => {
+                    console.log(error);
+                });
+        }
+	}
+
+	checkPermissions(){
+		if(this.props.session.admin){
```

Comment:
Should fix indentation here. Looks off here, not sure if it's because of difference in tab vs spaces.